pkg/kube/client/erased.go

```diff
@@ -0,0 +1,67 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	klabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/pilot/pkg/util/informermetric"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/controllers"
+	"istio.io/istio/pkg/ptr"
+	"istio.io/pkg/log"
+)
+
+// Erased is a Cached with
```

Comment:
nit: It is not clear what does Erased do?

Reply:
It's basically a non -generic type. Used in places where we don't know the types. The names I am very open to changing, I don't love them

Reply:
Also looks like I forgot to finish the comment here..

Reply:
I see. you mean some thing similar to type erased? I agree naming is little hard here.  Now I understand the diff between this and cached. I will think about naming a bit. I will review in detail tomorrow

Reply:
I think we could also just make it a Cached[controllers.Object] if its more clear... will probably do that

Reply:
I renamed to Untyped and just made it an alias for `Cached[controllers.Object]`